resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/proxy/ClientInvoker.java

```diff
@@ -117,26 +120,24 @@ protected ClientInvocation createRequest(Object[] args)
          }
       }
 
-      ClientInvocationBuilder builder = null;
+      ClientConfiguration parentConfiguration=(ClientConfiguration) target.getConfiguration();
+      ClientInvocation clientInvocation = new ClientInvocation(this.webTarget.getResteasyClient(), target.getUri(),
+    		  new ClientRequestHeaders(parentConfiguration), parentConfiguration);
       if (accepts != null)
       {
-         builder = (ClientInvocationBuilder)target.request(accepts);
+         clientInvocation.getHeaders().accept(accepts);
       }
-      else
-      {
-         builder = (ClientInvocationBuilder)target.request();
-      }
-
       for (int i = 0; i < processors.length; i++)
       {
          if (processors != null && processors[i] instanceof InvocationProcessor)
          {
             InvocationProcessor processor = (InvocationProcessor)processors[i];
-            processor.process(builder, args[i]);
+            processor.process(clientInvocation, args[i]);

```

Comment:
Ok, I can see that passing ClientInvocation rather than ClientInvocationBuilder is a little simpler, but I don't follow the statement "Further more, according to the way jaxrs Invocation.Builder is designed we can't built Invocation with entity in too steps contrary to what is done in ClientInvoker.createRequest(Object[]). If invocation need to be built with an entity it must be done in one step (see build(method,entity), method(method,entity)...)."

MessageBodyParameterProcessor is setting the entity in the invocation, right? What am I missing?